Sources/FoundationEssentials/Calendar/Calendar_Enumerate.swift

```diff
@@ -261,86 +279,182 @@ private func _handleCalendarResultNotFound(date: Date, calendar: Calendar, comps
+                        finished = true
                         return nil
+                    } catch {
+                        fatalError("Unexpected Calendar enumeration error type")
                     }
-                } catch (let e as CalendarEnumerationError) {
-                    _handleCalendarError(e, date: start, calendar: calendar, comps: matchingComponents, direction: direction, matchingPolicy: matchingPolicy, repeatedTimePolicy: repeatedTimePolicy)
+                } while true
+            }
+        }
+        
+        public func makeIterator() -> Iterator {
+            return Iterator(calendar, start: start, range: range, matching: matchingComponents, matchingPolicy: matchingPolicy, repeatedTimePolicy: repeatedTimePolicy, direction: direction)
+        }
+    }
+    
+    /// A `Sequence` of `Date`s, calculated by iterative addition of `DateComponents`.
+    struct DatesByAdding : Sendable, Sequence {
+        typealias Element = Date
+        
+        var calendar: Calendar
+        var start: Date
+        var range: Range<Date>?
+        var components: DateComponents?
+        var wrappingComponents: Bool
+        
+        /// If `components` is `nil`, the result is an empty `Sequence`.
+        internal init(calendar: Calendar, start: Date, range: Range<Date>?, components: DateComponents?, wrappingComponents: Bool) {
+            self.calendar = calendar
+            self.start = start
+            self.range = range
+            self.components = components
+            self.wrappingComponents = wrappingComponents
+        }
+        
+        struct Iterator: Sendable, IteratorProtocol {
+            var calendar: Calendar
+            var previousResult: Date?
+            let range: Range<Date>?
+            var components: DateComponents?
+            var wrappingComponents: Bool
+            
+            /// If `components` is `nil`, the result is an empty `Sequence`.
+            init(calendar: Calendar, start: Date, range: Range<Date>?, components: DateComponents?, wrappingComponents: Bool) {
+                self.calendar = calendar
+                previousResult = start
+                self.components = components
+                self.range = range
+                self.wrappingComponents = wrappingComponents
+            }
+            
+            mutating func next() -> Element? {
+                guard let components else {
+                    // We have nothing to add
+                    return nil
+                }
+                
+                guard let startAt = previousResult else {
+                    // We have finished
+                    return nil
+                }
+                                
+                let next = calendar.date(byAdding: components, to: startAt, wrappingComponents: wrappingComponents)
```

Comment:
This approach is susceptible to the "1+1≠2" problem in calendar math. By always adding from the previous result, we're allowing for drift to occur.

As a simple example, if the start date is "January 31" and the stride is "1 month", then the first addition will result in "February 28". The addition after that will result in "March 28", which has drifted from the likely expected value of "March 31".

The typical way to solve this is to always add from the start date, and to scale the value being added. So you start by adding one month, then two months, then three months etc; instead of adding one month every time.

This, however, gives rise to other questions about which units can be used as a stride.

`DateComponents` allows you to specify values that aren't relevant to addition, such as the `.calendar`, `.timezone`, `.weekday`, etc. It seems to me that the sequence and iterator initializers should include checks to `fatalError()` if the components include values that are illogical to use as a stride.